src/core/load-fragment.c

```diff
@@ -4606,7 +4606,7 @@ int config_parse_exec_directories(
                 if (r == -ENOMEM)
                         return log_oom();
                 if (r <= 0) {
-                        log_syntax(unit, LOG_WARNING, filename, line, r ?: SYNTHETIC_ERRNO(EINVAL),
+                        log_syntax(unit, LOG_WARNING, filename, line, r,
```

Comment:
isn't it a problem if `0` is passed as the error number?

Reply:
guess not

Reply:
Not in `log_syntax`.